pubsub/src/it/scala/com/snowplowanalytics/snowplow/collectors/scalastream/pubsub/GooglePubSubCollectorSpec.scala

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2022-2022 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0, and
+ * you may not use this file except in compliance with the Apache License
+ * Version 2.0.  You may obtain a copy of the Apache License Version 2.0 at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the Apache License Version 2.0 is distributed on an "AS
+ * IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the Apache License Version 2.0 for the specific language
+ * governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.collectors.scalastream.pubsub
+
+import scala.concurrent.duration._
+
+import cats.effect.IO
+
+import cats.effect.testing.specs2.CatsIO
+
+import org.specs2.mutable.Specification
+import org.specs2.specification.BeforeAfterAll
+
+import utils._
+
+class GooglePubSubCollectorSpec extends Specification with CatsIO with BeforeAfterAll {
+
+  override protected val Timeout = 5.minutes
+
+  def beforeAll: Unit = Containers.startEmulator()
+
+  def afterAll: Unit = Containers.stopEmulator()
+
+  "collector-pubsub" should {
+    "be able to parse the minimal config" in {
+      val testName = "minimal"
+      Containers.collector(
+        "examples/config.pubsub.minimal.hocon",
+        testName
+      ).use { collector =>
+        IO(collector.getLogs() must contain(("Setting health endpoint to healthy")))
+      }
+    }
+
+    "emit the correct number of enriched events and bad rows" in {
+      val testName = "count"
+      val nbGood = 1000
+      val nbBad = 10
+
+      Containers.collector(
+        "examples/config.pubsub.minimal.hocon",
+        testName,
+        Map("JDK_JAVA_OPTIONS" -> s"-Dcollector.streams.sink.maxBytes=$maxBytes")
+      ).use { collector =>
+        for {
+          _ <- log(testName, "Sending data")
+          _ <- EventGenerator.sendRequests(
+            collector.getHost(),
+            collector.getMappedPort(Containers.collectorPort),
+            nbGood,
+            nbBad
+          )
+          _ <- log(testName, "Data sent. Waiting for collector to work")
+          _ <- IO.sleep(5.second)
+          _ <- log(testName, "Consuming collector's output")
+          collectorOutput <- PubSub.consume(
+            Containers.projectId,
+            Containers.emulatorHost,
+            Containers.emulatorHostPort,
+            Containers.topicGood,
+            Containers.topicBad
+          )
+          _ <- IO(printBadRows(testName, collectorOutput.bad))
+        } yield {
+          collectorOutput.good.size should beEqualTo(nbGood)
+          collectorOutput.bad.size should beEqualTo(nbBad)
+        }
+      }
+    }
+
+    "shutdown when it receives a SIGTERM" in {
```

Comment:
Just thinking out loud... I wonder if this should be "shutdown within XX seconds when it receives a SIGTERM".  And then change to `Containers.waitUntilStopped(collector).timeout(10.seconds)`.